backend/pkg/server/schedule.go

```diff
@@ -6,12 +6,59 @@ import (
 	"net/http"
 
 	"github.com/gorilla/mux"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
 	"github.com/grafana/simple-datasource-backend/pkg/dbstore"
 )
 
+func (server *HttpServer) fetchSchedules(rw http.ResponseWriter, request *http.Request) {
+	var schedules []dbstore.Schedule
+
+	schedules, err := server.db.GetSchedules()
+	if err != nil {
+		log.DefaultLogger.Error("fetchSchedules: db.GetSchedules(): " + err.Error())
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = json.NewEncoder(rw).Encode(schedules)
+	if err != nil {
+		log.DefaultLogger.Error("fetchSchedules: json.NewEncoder().Encode(): " + err.Error())
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+
 func (server *HttpServer) createSchedule(rw http.ResponseWriter, request *http.Request) {
-	schedule, _ := server.db.CreateSchedule()
-	json.NewEncoder(rw).Encode(schedule)
+	schedule, err := server.db.CreateSchedule()
+	if err != nil {
+		log.DefaultLogger.Error("createSchedule: db.CreateSchedule(): " + err.Error())
+		http.Error(rw, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err = json.NewEncoder(rw).Encode(schedule)
+	if err != nil {
+		log.DefaultLogger.Error("createSchedule: json.NewEncoder().Encode(): " + err.Error())
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (server *HttpServer) deleteSchedule(rw http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars["id"]
```

Comment:
Is there a difference/preference between `vars["id"]`  and `vars.id` here?

Reply:
I only really like using dot notation when it is guaranteed camel case - url vars aren't .. I dunno its weird

Reply:
Ah yup, fair enough!